contracts/near/eth-prover/src/lib.rs

```diff
@@ -136,6 +136,94 @@ impl EthProver {
             .into()
     }
 
+    #[result_serializer(borsh)]
+    pub fn verify_account_proof(
+        &self,
+        #[serializer(borsh)] header_data: Vec<u8>,
+        #[serializer(borsh)] proof: Vec<Vec<u8>>, // account proof
+        #[serializer(borsh)] key: Vec<u8>,        // keccak256 of eth address
+        #[serializer(borsh)] account_data: Vec<u8>, // rlp encoded account state
+        #[serializer(borsh)] skip_bridge_call: bool,
+    ) -> PromiseOrValue<bool> {
+        self.check_not_paused(PAUSE_VERIFY);
+        let header: BlockHeader = rlp::decode(header_data.as_slice()).unwrap();
+        let data = Self::verify_trie_proof(header.state_root, key, proof);
+
+        let verification_result = data == account_data;
+        if verification_result && skip_bridge_call {
+            return PromiseOrValue::Value(true);
+        } else if !verification_result {
+            return PromiseOrValue::Value(false);
+        }
+        // Verify block header was in the bridge
+        eth_client::ext(self.bridge_smart_contract.parse().unwrap())
+            .with_static_gas(BLOCK_HASH_SAFE_GAS)
+            .block_hash_safe(header.number)
+            .then(
+                remote_self::ext(env::current_account_id())
+                    .with_static_gas(ON_BLOCK_HASH_GAS)
+                    .on_block_hash(header.hash.unwrap()),
+            )
+            .into()
+    }
+
+    #[result_serializer(borsh)]
+    pub fn verify_storage_proof(
+        &self,
+        #[serializer(borsh)] header_data: Vec<u8>,
+        #[serializer(borsh)] account_proof: Vec<Vec<u8>>, // account proof
+        #[serializer(borsh)] contract_address: Vec<u8>,   // eth address
+        #[serializer(borsh)] account_state: Vec<u8>,      // encoded account state
+        #[serializer(borsh)] storage_key: Vec<u8>,        // keccak256 of storage key
+        #[serializer(borsh)] storage_proof: Vec<Vec<u8>>, // storage proof
+        #[serializer(borsh)] expected_storage_value: Vec<u8>, // storage value
+        #[serializer(borsh)] min_header_height: Option<u64>,
+        #[serializer(borsh)] max_header_height: Option<u64>,
+        #[serializer(borsh)] skip_bridge_call: bool,
+    ) -> PromiseOrValue<bool> {
+        self.check_not_paused(PAUSE_VERIFY);
+        let header: BlockHeader = rlp::decode(header_data.as_slice()).unwrap();
+
+        
+        if let Some(min_header_height) = min_header_height {
+            if header.number < min_header_height {
+                return PromiseOrValue::Value(false);
+            }
+        }
+        
+        if let Some(max_header_height) = max_header_height {
+            if header.number > max_header_height {
+                return PromiseOrValue::Value(false);
```

Comment:
Is it possible somehow to log the error reason? If it is possible I think It will be extremely useful in future development.

Reply:
Added in [473709f](https://github.com/aurora-is-near/rainbow-bridge/pull/875/commits/473709f6b717b55f145419b5919f138ad1f2a92e), thanks for commenting out.